src/components/button/button.tsx

```diff
@@ -13,25 +15,41 @@ export type ButtonProps = {
   handleClick?: () => void;
   danger?: boolean;
   size?: ButtonSizes;
+  icon?: iconType;
+  iconOnRight?: boolean;
 } & StitchesButtonProps;
 
 const IonButton = ({
   type = 'primary',
   disabled = false,
   danger = false,
   size = 'md',
+  icon,
+  iconOnRight,
   label,
   handleClick,
 }: ButtonProps) => {
+  const iconSize = {
+    sm: 16,
+    md: 20,
+    lg: 24,
+    xl: 28,
+  };
+
   return (
     <ButtonStyles
       type={type}
       onClick={handleClick}
       disabled={disabled}
       danger={danger}
       size={size}
+      withIcon={icon && true}
```

Comment:
this true is really necessary?

Reply:
Yes. 'withIcon' is a variant that only activates if an icon exists. The property only accepts boolean. Do you have a better suggestion?

Reply:
!!icon